### DISCUSSION TITLE
osu!lazer fps locked to ~60 and inconsistent frametimes

### DISCUSSION THREAD

**Discussion by hostilew** [ppy/osu#15445] [Q&A] [answered]
windows 10
i7-8700k, gtx 1060 6gb, 32gb ram with latest nvidia drivers, 
dual screen monitor set-up with 2 different refresh rates and resolutions (1440p 165, 1080p 60)
not sure how to attach lazer logs

Game performance was perfect before Nov 2 update but after installing the update ingame fps is locked to ~60 on borderless fullscreen  fullscreen and windowed modes(game says draw is 240 but visually it looks like 60fps).  All modes have inconsistent frametimes(tested by drawing circles with my cursor, on desktop the cursor trails with perfect gaps but on lazer the gaps are inconsistent -frames can draw very close to each other or far apart.) There is also now a black screen whenever I enter fullscreen mode that takes a few seconds to go away. 

Things I tried that didn't change anything: multithreading vs singlethreading, 2x vs 4x vs unlimited refresh rates, disabling fullscreen optimizations, turning off g-sync.

[database.log](https://github.com/ppy/osu/files/7465391/database.log)
[network.log](https://github.com/ppy/osu/files/7465392/network.log)
[performance.log](https://github.com/ppy/osu/files/7465393/performance.log)
[performance-draw.log](https://github.com/ppy/osu/files/7465394/performance-draw.log)
[performance-update.log](https://github.com/ppy/osu/files/7465395/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/7465396/runtime.log)
[updater.log](https://github.com/ppy/osu/files/7465397/updater.log)

**Comment by peppy** [comment, marked as answer]
Did you try restarting your PC?

Also please logs: https://github.com/ppy/osu/discussions/15188

**Reply by peppy** [reply]
The logs provided are from stable.

Let me also preface this with: the issue is not going to be osu!. It's going to be another change on your PC.

**Reply by hostilew** [reply]
I'm not sure how to attach lazer logs

**Reply by peppy** [reply]
I literally linked the link to the link to the link liink lii

**Reply by hostilew** [reply]
thanks I was in wrong appdata

**Reply by peppy** [reply]
Anyways everything looks fine on lazer's side. Please troubleshoot using https://osu.ppy.sh/help/wiki/Performance_Troubleshooting.

**Reply by hostilew** [reply]
unplugging my second monitor fixed my problem

**Reply by peppy** [reply]
Did you have it unplugged previously? Alternatively, do you have something like discord open on your second monitor? And is it lower refresh rate than your main one? If so, that's a windows/nvidia driver limitation and will affect more than just osu!.

**Reply by hostilew** [reply]
Nah nothing is open on my second monitor and I don't have discord installed. My 2nd monitor was plugged in the whole time and it is lower refresh rate than my main monitor. I noticed lazer was drawing 240 frames when i had 4x frame selected, so lazer thinks i'm running 60fps when my second 60hz monitor is plugged in. My main monitor is 165hz. I don't know why this happens.

**Reply by peppy** [reply]
It could also be related to the exclusive fullscreen hack we are using (where one pixel is protruding one edge of the display). If your second monitor is to the left of your main one, try moving it to the right of below and see if it helps.

Basically, the issue arises when something GPU accelerated is drawing to the lower refresh rate monitor, causing the system-wide draw rate to drop to match it.

**Reply by hostilew** [reply]
Oh I noticed that there's a strip of pixels exposed on my left monitor I was wondering why that was there. I found another way to fix my problem was just starting the game in fullscreen mode already(exiting in fullscreen so that it opens in fs) and that seems to work. thanks for that knowledge though.

**Comment by hostilew** [comment]
Yes I also restarted my pc. Will update with recording
